tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2785,13 +2785,17 @@ AddBuiltInTriangleIntersectionAttributes(ASTContext &context,
 //
 // Subobjects
 
-static CXXRecordDecl *StartSubobjectDecl(ASTContext &context,
-                                         const char *name) {
+static CXXRecordDecl *
+StartSubobjectDecl(ASTContext &context, const char *name,
+                   DXIL::SubobjectKind kind,
+                   DXIL::HitGroupType hgtype = (DXIL::HitGroupType)(-1)) {
```

Comment:
I don't think we should cast an in valid enum value (-1) to the enum type.

Recommend using `DXIL::HitGroupType::LastEntry` instead.

You can use `IsValidHitGroupType()` to check if it has been supplied with a valid value when interpreting it later.

In the attribute definition, we do not need a default value, since it is always created with an explicit value.  This solves the problem of keeping the int representation in the `.td` file for the default value in sync with `LastEntry`.

Let's also name our new parameters based on our coding standard.

```suggestion
                   DXIL::SubobjectKind Kind,
                   DXIL::HitGroupType HGT = DXIL::HitGroupType::LastEntry) {
```